src/reps/object.js

```diff
@@ -1,173 +1,174 @@
-      // Do not pick objects, it could cause recursion.
-      return (t == "boolean" || t == "number" || (t == "string" && value));
-    };
-
-    // Work around https://bugzilla.mozilla.org/show_bug.cgi?id=945377
-    if (Object.prototype.toString.call(object) === "[object Generator]") {
-      object = Object.getPrototypeOf(object);
-    }
+ObjectRep.propTypes = {
+  object: React.PropTypes.object.isRequired,
+  // @TODO Change this to Object.values once it's supported in Node's version of V8
+  mode: React.PropTypes.oneOf(Object.keys(MODE).map(key => MODE[key])),
+  objectLink: React.PropTypes.func,
+  title: React.PropTypes.string,
+};
 
-    // Object members with non-empty values are preferred since it gives the
-    // user a better overview of the object.
-    let propsArray = this.getPropsArray(object, max, isInterestingProp);
-
-    if (propsArray.length <= max) {
-      // There are not enough props yet (or at least, not enough props to
-      // be able to know whether we should print "more…" or not).
-      // Let's display also empty members and functions.
-      propsArray = propsArray.concat(this.getPropsArray(object, max, (t, value) => {
-        return !isInterestingProp(t, value);
-      }));
-    }
+function ObjectRep(props) {
+  let object = props.object;
+  let propsArray = safePropIterator(props, object);
 
-    if (propsArray.length > max) {
-      propsArray.pop();
-      let objectLink = this.props.objectLink || span;
-
-      propsArray.push(Caption({
-        object: objectLink({
-          object: object
-        }, (Object.keys(object).length - max) + " more…")
-      }));
-    } else if (propsArray.length > 0) {
-      // Remove the last comma.
-      propsArray[propsArray.length - 1] = React.cloneElement(
-        propsArray[propsArray.length - 1], { delim: "" });
-    }
+  if (props.mode === MODE.TINY || !propsArray.length) {
+    return (
+      span({className: "objectBox objectBox-object"},
+        getTitle(props, object)
+      )
+    );
+  }
+
+  return (
+    span({className: "objectBox objectBox-object"},
+      getTitle(props, object),
+      safeObjectLink(props, {
+        className: "objectLeftBrace",
+      }, " { "),
+      ...propsArray,
+      safeObjectLink(props, {
+        className: "objectRightBrace",
+      }, " }")
+    )
+  );
+}
 
-    return propsArray;
-  },
+function getTitle(props, object) {
+  let title = props.title || object.class || "Object";
+  return safeObjectLink(props, {className: "objectTitle"}, title);
+}
 
-  getPropsArray: function (object, max, filter) {
-    let propsArray = [];
+function safePropIterator(props, object, max) {
+  max = (typeof max === "undefined") ? 3 : max;
+  try {
+    return propIterator(props, object, max);
+  } catch (err) {
+    console.error(err);
+  }
+  return [];
+}
 
-    max = max || 3;
-    if (!object) {
-      return propsArray;
-    }
+function propIterator(props, object, max) {
+  let isInterestingProp = (t, value) => {
```

Comment:
Maybe use an explicit argument name "type" instead of t (I know this is not really changed by your commit, but since you modified a bit the way we build the props arrays here, I had to read it carefully and the "t"s were pretty annoying :) )